src/lsp/protocol/server_capabilities.cr

```diff
@@ -19,41 +17,39 @@ module LSP::Protocol
     end
   end
 
-  # Specify Sever capabilities supported,
-  # Currently Crystal supports:
+  # Specify supported server capabilities:
   #
-  # - Go to definition or Implementation (WIP)
+  # - Go to Implementation (WIP)
   # - Diagnostics (WIP)
   # - Formatting (WIP)
-  # - Symbols https://github.com/crystal-lang/crystal/blob/1f3e8b0e742b55c1feb5584dc932e87034365f48/samples/compiler/visitor_example.cr
-  # - Rename https://github.com/crystal-lang/crystal/blob/1f3e8b0e742b55c1feb5584dc932e87034365f48/samples/compiler/transformer_example.cr
-  # - Completion https://github.com/TechMagister/cracker
-  # - Hover
-  # - Code Actions
-  # - Signature Help
-  # - Range Formatting
+  # - Symbols (WIP)
+  # - Completion (WIP)
+  # - Hover (WIP)
+  # - Code Actions (WIP)
   #
   # Features not supported by Crystal yet:
   #
-  # - CodeLens
+  # - Rename
   # - Find References
-  #
-  # To enable more capabilities, See: https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md
   struct ServerCapabilities
     JSON.mapping(
       textDocumentSync: TextDocumentSyncKind,
       documentFormattingProvider: Bool,
       definitionProvider: Bool,
       documentSymbolProvider: Bool,
-      completionProvider: CompletionOptions
+      workspaceSymbolProvider: Bool,
+      completionProvider: CompletionOptions,
+      hoverProvider: Bool,
     )
 
     def initialize
       @textDocumentSync = TextDocumentSyncKind::Full
       @documentFormattingProvider = true
       @definitionProvider = true
       @documentSymbolProvider = true
+      @workspaceSymbolProvider = true
       @completionProvider = CompletionOptions.new
+      @hoverProvider = true
```

Comment:
One last thing and it'll be good to (test &) go! 

The server capabilities is a scry knowledge, can you move the initialization of this struct to Scry? :pray: